src/Facebook/InstantArticles/Validators/InstantArticleValidator.php

```diff
@@ -24,29 +24,28 @@ class InstantArticleValidator
      * This method navigates thru the tree structure and validates the article content.
      *
      * @param InstantArticle $article The article that will be checked.
-     * @return array of string with the warnings raised during the check.
+     * @return vec of string with the warnings raised during the check.
      */
-    public static function check(InstantArticle $article): Vector<ValidatorWarning>
+    public static function check(InstantArticle $article): vec<ValidatorWarning>
     {
-        $warnings = Vector {};
-        return self::getReport($article->getContainerChildren(), $warnings);
+        return self::getReport($article->getContainerChildren(), vec[]);
     }
 
     /**
      * Auxiliary method to do a recursive checker that will raise all warnings
      * related to the element tree about the Instant Article.
-     * @param array $elements Element[] to all elements that will be checked.
-     * @param array $warnings string[] to all warnings related to the elements informed.
+     * @param vec $elements Element[] to all elements that will be checked.
+     * @param vec $warnings string[] to all warnings related to the elements informed.
      */
-    public static function getReport(Vector<Element> $elements, Vector<ValidatorWarning>$warnings): Vector<ValidatorWarning>
+    public static function getReport(vec<Element> $elements, vec<ValidatorWarning>$warnings): vec<ValidatorWarning>
     {
         foreach ($elements as $element) {
             if (!$element->isValid() && $element->isEmptyValidationEnabled()) {
                 // Adds a warning to the result report.
-                $warnings->add(new ValidatorWarning($element));
+                $warnings[] = new ValidatorWarning($element);
             }
             if ($element instanceof ChildrenContainer) {
-                self::getReport($element->getContainerChildren(), $warnings);
+                $warnings = Type::concatVec($warnings, self::getReport($element->getContainerChildren(), vec[]));
```

Comment:
Can't you use the existing Vec\concat() function here?

Reply:
First run was with few issues then I dropped and forgot to re-add it. Good call.
Pushed.